tests/_support/Helper/AsyncApiHelper.php

```diff
@@ -389,4 +387,72 @@ public function getAsyncApiBuilderConsoleMock(): BuildCodeFromAsyncApiConsole
 
         return $buildFromAsyncApiConsole;
     }
+
+    /**
+     * @return \Generated\Shared\Transfer\AsyncApiMessageTransfer
+     */
+    public function haveAsyncApiMessagePropertyRequestTransfer(): AsyncApiMessageTransfer
+    {
+        $asyncApiChannelTransfer = new AsyncApiChannelTransfer();
+        $asyncApiChannelTransfer->setName(static::CHANNEL_NAME);
+
+        $asyncApiMessageTransfer = new AsyncApiMessageTransfer();
+        $asyncApiMessageTransfer
+            ->setChannel($asyncApiChannelTransfer)
+            ->setName('message')
+            ->setProperty(['firstName:string:required', 'lastName:string', 'phoneNumber:int:required'])
+            ->setContentType('object');
+
+        return $asyncApiMessageTransfer;
+    }
+
+    /**
+     * @param string $targetFile
+     * @param string $channelName
+     * @param string $messageName
+     * @param array $property
+     *
+     * @return void
+     */
+    public function assertChannelProperty(string $targetFile, string $channelName, string $messageName, array $property): void
+    {
+        $asyncApi = Yaml::parseFile($targetFile);
+
+        $this->assertArrayHasKey($channelName, $asyncApi['components'], sprintf(
+            'Expected to have a channel name "%s" but it does not exist.',
+            $channelName,
+        ));
+
+        $this->assertArrayHasKey($messageName, $asyncApi['components'][$channelName], sprintf(
+            'Expected to have a "%s" message name in the channel "%s" but it does not exist.',
+            $channelName,
+            $messageName,
+        ));
+
+        $this->assertArrayHasKey($property[0], $asyncApi['components'][$channelName][$messageName]['properties'], sprintf(
+            'Expected to have a property type "%s" in the message name "%s" but it does not exist.',
+            $property[0],
+            $messageName,
+        ));
+
+        $this->assertSame($asyncApi['components'][$channelName][$messageName]['properties'][$property[0]]['type'], $property[1], sprintf(
+            'Expected to have a "%s" property type in message "%s" but it does not exist.',
+            $property[1],
+            $messageName,
+        ));
+
+        if (in_array($property[0], $asyncApi['components'][$channelName][$messageName]['required'])) {
+            $this->assertTrue($property[2], sprintf(
+                'Expected to have a "%s" property type in "%s" required but it does not exist.',
+                'required',
+                $property[0],
+            ));
+        } else {
+            $this->assertFalse($property[2], sprintf(
+                'Expected to have a "%s" property type optional in "%s" but it does not exist.',
+                'Optional',
```

Comment:
this message doesn't make any sense to me, and I don't see any reason to have the word "Optional" as a separate parameter

Reply:
I removed this extra parameter and used it in the message string.